agent/Agent.js

```diff
@@ -96,22 +98,34 @@ class Agent extends EventEmitter {
   capabilities: {[key: string]: boolean};
   _updateScroll: () => void;
   _inspectEnabled: boolean;
+  _prevInspectedHooks: any = null;
 
   constructor(global: Object, capabilities?: Object) {
     super();
     this.global = global;
     this.internalInstancesById = new Map();
     this.idsByInternalInstances = new WeakMap();
+    this.rendererIdsByInternalInstance = new WeakMap();
     this.renderers = new Map();
     this.elementData = new Map();
     this.roots = new Set();
     this.reactInternals = {};
     var lastSelected;
     this.on('selected', id => {
       var data = this.elementData.get(id);
-      if (data && data.publicInstance && this.global.$r === lastSelected) {
-        this.global.$r = data.publicInstance;
-        lastSelected = data.publicInstance;
+      var inspectedHooks = null;
+      if (data) {
+        if (data.publicInstance && this.global.$r === lastSelected) {
+          this.global.$r = data.publicInstance;
+          lastSelected = data.publicInstance;
+        }
+        if (data.containsHooks) {
+          inspectedHooks = this.updateHooksTree(id);
+        }
+      }
+      if (this._prevInspectedHooks !== inspectedHooks) {
+        this._prevInspectedHooks = inspectedHooks;
+        this.emit('inspectedHooks', inspectedHooks);
```

Comment:
Doing this (storing data, having Fiber-specific code) in the Agent sucks. I'm open to suggestions for a better place for this logic to live, but at the same time– I didn't stress about it too much because I think this version of DevTools is short lived (without either a rewrite or a refactor).